### PR TITLE
Gun stabilizer/recoil compensator

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Used to upgrade guns. Removes their heavy_weapon and recoil, allowing you to fire them one handed.

10tc in the ops uplink

:cl: Kor
rscadd: A gun stabilizer has been added to the nuke ops uplink. Attaching it to a will remove any recoil snd let you fire the gun one handed
/:cl:

### CODE REVIEW

**Comment by Jordie0608** [comment]
Although I can't imagine it'd be used much a way to remove would be good since it'll eventually come up.

**Comment by KazeEspada** [comment]
Dual wielding saws.

File: code/modules/projectiles/gun_upgrades.dm [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/obj/item/device/recoil_compensator
+	name = "anti-grav stabilizer"
+	desc = "A miniaturized gravity generator, capable of reducing the weight of a gun as well as reducing recoil."
+	icon = 'icons/obj/stock_parts.dmi'
+	icon_state = "quadratic_capacitor"
+
+/obj/item/device/recoil_compensator/afterattack(obj/item/weapon/gun/G, mob/user)
```
Comment (tkdrg) [inline code/modules/projectiles/gun_upgrades.dm]: Needs a check for `proximity` or perhaps for `G in user.contents`. Also I think `!user.incapacitated()` (not sure if click code handles that already but probably not for afterattack).

**Comment by tkdrg** [comment]
@KorPhaeron

**Comment by KorPhaeron** [comment]
Don't really want to add steps for removing this right now, maybe this can exist again when goof does his gunsmithing thing

**Comment by Shapsy** [comment]
@korPhaeron couldn't you just steal how the egun works with the seclite and use a screwdriver to detach it?

**Comment by KorPhaeron** [comment]
I guess I could look at that. I have too many PRs open/am feeling stretched thin though, I would rather focus on the three big lavaland ones I have open

**Pull request closed without merging**